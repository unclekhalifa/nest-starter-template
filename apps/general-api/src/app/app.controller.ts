import { Body, Controller, Get, Post } from '@nestjs/common';

import { AppService } from './app.service';

@Controller()
export class AppController {
  constructor(private readonly appService: AppService) {}

  @Get()
  getData() {
    return this.appService.getData();
  }

  @Post()
  createUser(@Body() body: any) {
    return this.appService.createUser(body.email);
  }
}
