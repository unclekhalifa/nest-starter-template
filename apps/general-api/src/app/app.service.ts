import { Injectable } from '@nestjs/common';
import { DatabaseService } from '../database';

@Injectable()
export class AppService {
  constructor(private readonly database: DatabaseService) {}

  getData(): { message: string } {
    return { message: 'Hello API' };
  }

  async createUser(email: string) {
    const User = await this.database.user.create({ data: { email: 'John@mail.com' } });
    return User;
  }
}
