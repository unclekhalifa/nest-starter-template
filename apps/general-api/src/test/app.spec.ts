import { INestApplication } from '@nestjs/common';
import { Test } from '@nestjs/testing';
import { AppModule } from '../app.module';
import request from 'supertest';

const createMock = jest.fn();
jest.mock('@prisma/client', () => {
  return {
    PrismaClient: jest.fn().mockImplementation(() => {
      return {
        user: {
          create: createMock,
        },
      };
    }),
  };
});

describe('General API', () => {
  let app: INestApplication;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let server: any;

  async function setup() {
    const module = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = module.createNestApplication();
    app.setGlobalPrefix('api');

    await app.init();
    server = app.getHttpServer();
  }

  beforeAll(async () => {
    await setup();
  });

  afterAll(async () => {
    await app.close();
  });

  describe('GET /', () => {
    const userMock = {
      id: 1,
      name: 'John Doe',
      email: 'john.doe@test.com',
    };
    it('should return 201', async () => {
      createMock.mockResolvedValue(userMock);
      const response = await request(server).post('/api/users')
        .send(userMock);
      expect(response.status).toBe(201);
      expect(response.body).toMatchSnapshot();
    });
  });

});
