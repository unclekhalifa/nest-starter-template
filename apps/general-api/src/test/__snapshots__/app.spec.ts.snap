// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General API GET / should return 201 1`] = `
{
  "email": "john.doe@test.com",
  "id": 1,
  "name": "John Doe",
}
`;
